sysdeps/linux/x86/crt-src/Scrt1.S

```diff
@@ -0,0 +1,33 @@
+.section .text
+
+// extern "C" [[noreturn, gnu::visibility("hidden"), gnu::weak]] void __stack_chk_fail_local() {
+// 	__stack_chk_fail();
+// }
+
+.type __stack_chk_fail_local, %function
+.weak __stack_chk_fail_local
+__stack_chk_fail_local:
```

Comment:
Why is this required?

Reply:
GCC sometimes emits calls to this; this seems to be required for PIC code. To quote bionic:

> __stack_chk_fail_local is wrapper for __stack_chk_fail. Compiler generates wrapper calls instead for PIC code only and only on IA32 for optimization purpose (see gcc/config/i386/i386.c). Wrapper body is always included into executable or library. This is the idea of optimization.

Reply:
But we already have the function defined in ``stack_protector.cpp``, why is it needed in crt1 as well?

Reply:
Ah, good catch. I've tested this locally and it doesn't seem to be required.

Reply:
Interestingly, GHA complains about it missing.

Reply:
> Interestingly, GHA complains about it missing.

On a clean version of mlibc also can reproduce it.

Reply:
Maybe an issue in the environment.

Reply:
> > Interestingly, GHA complains about it missing.
> 
> It's not entirely GHA issue. Can also reproduce it on Ubuntu 22.04.1 with a clean version of mlibc.

[3c9a485](https://github.com/managarm/mlibc/pull/702/commits/3c9a485407fc5a465a71b402a42fd0c80dcca069) Fixed it.

Reply:
Not sure if I consider resolved, why does it happen on some platforms and not on others?